app/assets/javascripts/govuk_publishing_components/load-analytics.js

```diff
@@ -0,0 +1,261 @@
+      window.GOVUK.analyticsVars.gaProperty = 'UA-26179049-22'
+    } else {
+      window.GOVUK.analyticsVars.gaPropertyCrossDomain = 'UA-UNSET'
+    }
+
+    // Load universal analytics
+    if (typeof window.GOVUK.analyticsInit !== 'undefined') {
+      window.GOVUK.analyticsInit()
+    }
+  },
+
+  loadGa4: function (currentDomain) {
+    currentDomain = currentDomain || window.location.hostname
+    var environment = ''
+
+    // Categorise current environment and set a couple UA variables
+    if (this.arrayContains(currentDomain, this.productionDomains)) {
+      environment = 'production'
+    } else if (this.arrayContains(currentDomain, this.stagingDomains)) {
+      environment = 'staging'
+    } else if (this.arrayContains(currentDomain, this.integrationDomains)) {
+      environment = 'integration'
+    } else if (this.arrayContains(currentDomain, this.developmentDomains) || currentDomain.indexOf('.dev.gov.uk') !== -1) {
+      environment = 'development'
+    }
+
+    // If we recognise the environment (i.e. the string isn't empty), load in GA4
+    if (environment) {
+      // If analytics-ga4.js exists and our detected environment has 'initialiseGA4' set to true, load GA4.
+      if (typeof window.GOVUK.analyticsGa4.init !== 'undefined' && this.ga4EnvironmentVariables[environment].initialiseGA4) {
+        window.GOVUK.analyticsGa4 = window.GOVUK.analyticsGa4 || {}
+        window.GOVUK.analyticsGa4.vars = window.GOVUK.analyticsGa4.vars || {}
+        window.GOVUK.analyticsGa4.vars.id = this.ga4EnvironmentVariables.id
+        window.GOVUK.analyticsGa4.vars.auth = this.ga4EnvironmentVariables[environment].auth
+        window.GOVUK.analyticsGa4.vars.preview = this.ga4EnvironmentVariables[environment].preview
+        window.GOVUK.analyticsGa4.vars.environment = environment // Used for testing and debugging
+
+        var gemVersion = document.querySelector('meta[name="govuk:components_gem_version"]')
+        if (gemVersion) {
+          gemVersion = gemVersion.getAttribute('content')
+        } else {
+          gemVersion = 'not found'
+        }
+
+        window.GOVUK.analyticsGa4.vars.gem_version = gemVersion
+        window.GOVUK.analyticsGa4.init()
+      }
+    }
+  },
+
+  arrayContains: function (valueToFind, array) {
+    return array.indexOf(valueToFind) !== -1
+  },
+
+  ga4EnvironmentVariables: {
+    id: 'GTM-MG7HG5W',
+    production: {
+      initialiseGA4: true
+    },
+    staging: {
+      initialiseGA4: true,
+      auth: 'oJWs562CxSIjZKn_GlB5Bw',
+      preview: 'env-5'
+    },
+    integration: {
+      initialiseGA4: true,
+      auth: 'C7iYdcsOlYgGmiUJjZKrHQ',
+      preview: 'env-4'
+    },
+    development: {
+      initialiseGA4: true,
+      auth: 'bRiZ-jiEHtw6hHpGd6dF9w',
+      preview: 'env-3'
+    }
+  }
+}
+
+// This needs to run in a DOMContentLoaded function otherwise tracking code crashes due to the <body> not existing.
+window.addEventListener('DOMContentLoaded', function () {
```

Comment:
I'm slightly hesitant over the use of `DOMContentLoaded` here as making this work reliably on older browsers is a bit of a juggling act. Having said that, this seems to work fine in IE9 and 11 (can't currently check 10 for some browserstack reason).

We do already have a DOMContentLoaded call in https://github.com/alphagov/govuk_publishing_components/blob/main/app/assets/javascripts/govuk_publishing_components/dependencies.js so I'm wondering if it's worth combining the two? I have a long standing issue to resolve to remove that one, so putting them together would help with that. No worries if this is too complex.

Reply:
Thanks @andysellick - I've moved it there now.